spec/services/creators/group_creator_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe GroupCreator do
+  context 'success' do
+    it 'sucessfully creates a group' do
+      params = instance_spy('ActionController::Parameters', to_h: params_hash)
+      expect(described_class.new(params).create![:group]).to \
+        be_instance_of(Group)
+    end
+    it 'returns a notice flash message' do
+      params = instance_spy('ActionController::Parameters', to_h: params_hash)
+      expect(described_class.new(params).create![:msg]).to have_key(:notice)
+    end
+    it 'sets :object to the draw and the new group' do
+      params = instance_spy('ActionController::Parameters', to_h: params_hash)
+      expect(described_class.new(params).create![:object].map(&:class)).to \
+        eq([Draw, Group])
+    end
+  end
+
+  it 'does not create when given invalid params' do
+    params = instance_spy('ActionController::Parameters', to_h: {})
+    expect(described_class.new(params).create![:object]).to be_nil
+  end
+
+  def params_hash
+    draw = FactoryGirl.create(:draw_with_members, students_count: 1)
```

Comment:
Do we want to memoize these just in case this ever gets called twice? I know it's unlikely but seeing guaranteed resource creation within a helper method is a little troubling to me.